### PR TITLE
New Discovery-Server SUPER_CLIENT participant type [10607]

### PR DESCRIPTION
**Pull request by jparisu**
Tests 32 & 33 : eProsima/Discovery-Server#24
Documentation: eProsima/Fast-DDS-docs#242

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3797/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/958/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3341/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3823/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3825/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/982/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3367/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3851/)

File: src/cpp/rtps/builtin/discovery/database/DiscoveryParticipantChangeData.hpp [hunk 1]
```diff
@@ -82,7 +82,9 @@ class DiscoveryParticipantChangeData
 
     // The metatraffic locators of from the serialized payload
     fastrtps::rtps::RemoteLocatorList metatraffic_locators_;
-    // Whether this participant is a CLIENT or a SERVER
+    // Whether this participant is a CLIENT or a SERVER/BACKUP/SUPER_CLIENT
+    // Finally, this variable affects to the discovery filter to apply to each entity:
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryParticipantChangeData.hpp:86]:
```suggestion
    // This variable affects the discovery filter applied to each entity:
```

File: test/unittest/xmlparser/XMLElementParserTests.cpp [hunk 2]
```diff
@@ -3249,6 +3249,13 @@ TEST_F(XMLParserTests, getXMLEnum_positive)
         titleElement = xml_doc.RootElement();
         EXPECT_EQ(XMLP_ret::XML_OK, XMLParserTest::getXMLEnum_wrapper(titleElement, &e, ident));
         EXPECT_EQ(DiscoveryProtocol_t::BACKUP, e);
+
```
Comment (JLBuenoLopez-eProsima) [inline test/unittest/xmlparser/XMLElementParserTests.cpp:3252]: The test comment should include that this case is also tested.

**Review by JLBuenoLopez-eProsima — COMMENTED** [review]
This feature should have a related documentation PR previous to being merged.

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM with green CI and related documentation PR

**Pull request merged**